src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/listeners/CollectFailedTestLogsListener.groovy

```diff
@@ -0,0 +1,59 @@
+package org.openkilda.functionaltests.listeners
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.spockframework.runtime.model.ErrorInfo
+import org.spockframework.runtime.model.FeatureInfo
+import org.spockframework.runtime.model.IterationInfo
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.client.RestTemplate
+
+import java.time.ZoneOffset
+
+import static java.time.LocalDateTime.now
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME
+
+class CollectFailedTestLogsListener extends AbstractSpringListener{
+    private String featureStartTime
+    private String iterationStartTime
+    @Value('${elasticsearch.endpoint}')
+    private String elasticSearchEndpoint
+    @Value('${elasticsearch.index}')
+    private String elasticSearchIndex
+
+
+    @Override
+    void beforeFeature(FeatureInfo feature) {
+        featureStartTime = utcTimeNow()
+    }
+
+    @Override
+    void beforeIteration(IterationInfo iterationInfo) {
+        iterationStartTime = utcTimeNow()
+    }
+
+    @Override
+    void error(ErrorInfo error) {
+        def objectMapper = new ObjectMapper()
+        def startTime = isIteration(error.getMethod().getIteration()) ? iterationStartTime : featureStartTime
+        def logs = new RestTemplate().getForEntity(
+                "${elasticSearchEndpoint}${elasticSearchIndex}/_search?q=" +
+                        "@timestamp:[${startTime} TO ${utcTimeNow()}]", String.class).getBody()
+        def beautifiedString = objectMapper.readValue(logs, Object.class)
+        objectMapper.writerWithDefaultPrettyPrinter().writeValue(getTargetLogFile(error), beautifiedString)
+    }
+
+    private static String utcTimeNow() {
+        return now(ZoneOffset.UTC).format(ISO_DATE_TIME)
+    }
+
+    private static File getTargetLogFile(ErrorInfo error) {
+        def specGroupName = error.getMethod().getParent().getPackage().split('\\.').last()
+        def specName = error.getMethod().getParent().getName()
+        def iterationName = error.getMethod().getIteration().getDisplayName()
+        return new File("build/logs/spec/${specGroupName}/${specName}/${iterationName}.server.log.json")
```

Comment:
Please, let me know it there's a way to retrieve gradle's build directory root in runtime.

Reply:
Not sure how to use gradle API from here, but you can save something as a system property in gradle.build and then read it here. For example, in a random test I had opened it looks like this:
```java
getClass().getClassLoader().getResourceAsStream("topology.properties")
```
then if you write sources root to a property, then you can get it here back.
But maybe this is overkill for this case.

Reply:
Hardcoding would be easier, I'm afraid.